airflow/operators/s3_to_hive_operator.py

```diff
@@ -105,28 +114,41 @@ def __init__(
         self.wildcard_match = wildcard_match
         self.hive_cli_conn_id = hive_cli_conn_id
         self.s3_conn_id = s3_conn_id
+        self.input_compressed = input_compressed
+
+        if (self.check_headers and
+                not (self.field_dict is not None and self.headers)):
+            raise AirflowException("To check_headers provide " +
+                                   "field_dict and headers")
 
     def execute(self, context):
-        self.hive = HiveCliHook(hive_cli_conn_id=self.hive_cli_conn_id)
+        # Downloading file from S3
         self.s3 = S3Hook(s3_conn_id=self.s3_conn_id)
+        self.hive = HiveCliHook(hive_cli_conn_id=self.hive_cli_conn_id)
         logging.info("Downloading S3 file")
+
         if self.wildcard_match:
             if not self.s3.check_for_wildcard_key(self.s3_key):
-                raise AirflowException("No key matches {0}".format(self.s3_key))
+                raise AirflowException("No key matches {0}"
+                                       .format(self.s3_key))
             s3_key_object = self.s3.get_wildcard_key(self.s3_key)
         else:
             if not self.s3.check_for_key(self.s3_key):
                 raise AirflowException(
                     "The key {0} does not exists".format(self.s3_key))
             s3_key_object = self.s3.get_key(self.s3_key)
-        with NamedTemporaryFile("w") as f:
+        root, file_ext = os.path.splitext(s3_key_object.key)
```

Comment:
nit: you might want to use `file_ext.lower()` in the rest of the code. I can totally see uppercase keys happening.